metricbeat/module/kibana/rules/data.go

```diff
@@ -0,0 +1,124 @@
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package rules
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/metricbeat/helper/elastic"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	s "github.com/elastic/beats/v7/libbeat/common/schema"
+	c "github.com/elastic/beats/v7/libbeat/common/schema/mapstriface"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+)
+
+var (
+	kibanaSchema = s.Schema{
+		"uuid":  c.Str("uuid"),
+		"name":  c.Str("name"),
+		"index": c.Str("index"),
+		"host": s.Object{
+			"name": c.Str("host"),
+		},
+		"transport_address": c.Str("transport_address"),
+		"version":           c.Str("version"),
+		"snapshot":          c.Bool("snapshot"),
+		"status":            c.Str("status"),
+	}
+
+	rulesSchema = s.Schema{
+		"overdue": c.Dict("overdue", s.Schema{
+			"count": c.Int("count"),
+			"duration": c.Dict("duration", s.Schema{
+				"p50": c.Float("p50"),
+				"p99": c.Float("p99"),
+			}),
+		}),
+		"failures":   c.Int("failures"),
+		"executions": c.Int("executions"),
+	}
+)
+
+type response struct {
+	Rules       map[string]interface{} `json:"rules"`
+	Kibana      map[string]interface{} `json:"kibana"`
+	ClusterUuid string                 `json:"cluster_uuid"`
+}
+
+func eventMapping(r mb.ReporterV2, content []byte, isXpack bool) error {
+	var data response
+	err := json.Unmarshal(content, &data)
+	if err != nil {
+		return errors.Wrap(err, "failure parsing Kibana Rules API response")
+	}
+
+	event := mb.Event{ModuleFields: common.MapStr{}, RootFields: common.MapStr{}}
+
+	// Set elasticsearch cluster id
+	event.ModuleFields.Put("elasticsearch.cluster.id", data.ClusterUuid)
+
+	kibana, _ := kibanaSchema.Apply(data.Kibana)
+	if err != nil {
+		event.Error = elastic.MakeErrorForMissingField("kibana", elastic.Kibana)
+		return event.Error
+	}
+
+	// Set service ID
+	serviceId, err := kibana.GetValue("uuid")
+	if err != nil {
+		event.Error = elastic.MakeErrorForMissingField("kibana.uuid", elastic.Kibana)
+		return event.Error
+	}
+	event.RootFields.Put("service.id", serviceId)
+
+	// Set service version
+	version, err := kibana.GetValue("version")
+	if err != nil {
+		event.Error = elastic.MakeErrorForMissingField("kibana.version", elastic.Kibana)
+		return event.Error
+	}
+	event.RootFields.Put("service.version", version)
+
+	// Set service address
+	serviceAddress, err := kibana.GetValue("transport_address")
+	if err != nil {
+		event.Error = elastic.MakeErrorForMissingField("kibana.transport_address", elastic.Kibana)
+		return event.Error
+	}
+	event.RootFields.Put("service.address", serviceAddress)
+
+	rulesFields, err := rulesSchema.Apply(data.Rules)
+	if err != nil {
```

Comment:
You probably found some example where we handle those errors. But I think that in general it's better to ignore them. It really depends if you want "partial data" or an "all or nothing". If one single field is missing, you get nothing, even if the rest of the fields are there. That's why it might be worth to ignore the error.